components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -573,12 +575,22 @@ nsapi_error_t ESP8266::send(int id, const void *data, uint32_t amount)
             if (_serial_rts == NC) {
                 while (_parser.process_oob()); // Drain USART receive register
             }
+            _busy = false;
+            _busy_timeout_reached = false;
             _smutex.unlock();
             return NSAPI_ERROR_OK;
         }
+        tr_debug("sending failed %d", i);
         if (_error) {
             _error = false;
         }
+        if (_busy && _busy_timeout_reached) {
+            tr_debug("returning NSAPI_ERROR_WOULD_BLOCK");
+            _busy = false;
```

Comment:
Unnecessary as each who is checking for busy must set this to false before making a request

Reply:
Otherwise could follow the same pattern as with
`nsapi_error_t ESP8266::connect(const char *ap, const char *passPhrase)`